From da5b0f9d385fa5388e6004f7c0a1e3072716394a Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 31 Mar 2022 20:15:02 +0300
Subject: [PATCH] [SW-77251] common: add wfe macro

This patch adds wfe macro which will put ARC to sleep

Change-Id: I9b57a846d5f613b2681f71a8066e9b37744cf469
---

diff --git a/include/arcboot.h b/include/arcboot.h
index 9babb18..ed4a1ec 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -17,6 +17,9 @@
 #include "hl_boot_desc_rom.h"
 #include "hl_sec_if.h"
 
+/* TODO: check whether below command puts ARC to sleep */
+#define wfe()	__asm__ __volatile__("sleep 0x11\n" : : )
+
 #ifndef __weak
 #define __weak		__attribute__((weak))
 #endif
